pkg/tarmak/cluster/cluster.go

```diff
@@ -203,7 +205,30 @@ func (c *Cluster) Validate() (result error) {
 		}
 	}
 
-	return result
+	// validate enforcement of instance pool count
+	if err := c.validateInstancePoolMultiple(); err != nil {
+		result = multierror.Append(result, err)
+	}
+
+	return result.ErrorOrNil()
+}
+
+func (c *Cluster) validateInstancePoolMultiple() error {
+	var result *multierror.Error
+
+	list := make(map[string][]*clusterv1alpha1.InstancePool)
+	for _, i := range c.Config().InstancePools {
+		list[i.Name] = append(list[i.Name], &i)
+	}
+
+	for name, v := range list {
+		if name != "worker" && len(v) > 1 {
+			err := fmt.Errorf("'%s' only supports a single instance pool", name)
```

Comment:
Maybe change message to "Instance pool type '%s' only supports a single instance pool"

Reply:
done